src/shared/clients/ssmClient.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AWSError, SSM } from 'aws-sdk'
+import globals from '../extensionGlobals'
+import { PromiseResult } from 'aws-sdk/lib/request'
+import { getLogger } from '../logger/logger'
+
+export class DefaultSsmClient {
+    public constructor(public readonly regionCode: string) {}
+
+    private async createSdkClient(): Promise<SSM> {
+        return await globals.sdkClientBuilder.createAwsService(SSM, undefined, this.regionCode)
+    }
+
+    public async terminateSession(
+        session: SSM.Session
+    ): Promise<void | PromiseResult<SSM.TerminateSessionResponse, AWSError>> {
+        const sessionId = session.SessionId!
+        const client = await this.createSdkClient()
+        const termination = await client
+            .terminateSession({ SessionId: sessionId })
+            .promise()
+            .catch(err => {
+                getLogger().warn(`ssm: failed to terminate session "${sessionId}": %s`, err)
+            })
+        return termination
+    }
+
+    public async startSession(
+        target: string,
+        callback?: ((err: AWSError, data: SSM.StartSessionResponse) => void) | undefined
```

Comment:
Prefer `async/await` over callbacks.

Reply:
I was having trouble detecting if it was an error or not, but just changed to `async/await` and checking `response.$response.error` which seems to work. Is that a reliable way to check if the response is an AWSError or is there easier/better way?

Reply:
Use `isAwsError` from `src/shared/errors.ts` in a catch block. If there's an error it should throw, are you using `.promise()`? Sorry forgot to mention that.

Reply:
Yeah, that's exactly what I ended up doing. I saw a few different examples of handling SDK errors in the code base so wasn't sure which to follow, but this seemed to be most prominent.